pysrc/bytewax/connectors/files.py

```diff
@@ -99,8 +111,33 @@ def build_part(self, for_part, resume_state):
         # TODO: Warn and return None. Then we could support
         # continuation from a different file.
         assert for_part == str(self._path), "Can't resume reading from different file"
-        return _FileSource(self._path, resume_state)
+        if self.file_type == '.csv':
+            return _CSVSource(self._path, resume_state)
+        else:
+            return _FileSource(self._path, resume_state)
+
+class _CSVSource(StatefulSource):
+    def __init__(self, path, resume_state):
+        resume_offset = resume_state or 0
+        self._f = open(path, "rt")
+        self.header = list(csv.reader([self._f.readline()]))[0]
+        header_position = self._f.tell()
+        if resume_offset <= header_position:
```

Comment:
The bit to pull out the header is needed to construct the dictionary? or the part below that sets resume_offset is probably not necessary since the file object isn't closed.

Reply:
I think you are right, it isn't necessary. I was envisioning a situation that will never occur when I think about it again.